### PR TITLE
ROC-7032 make mediation agreement mandatory ccd field

### PR DESCRIPTION
**Pull request by Valkrys**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-7522

### Change description ###
Make mediation agreement upload by staff on ccd mandatory

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/MediationSuccessfulRule.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4957521&bid=16382220)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/MediationSuccessfulRule.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4957521&bid=16382220)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/MediationSuccessfulRuleTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/MediationSuccessfulRule.java  11
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4957521&bid=16382220)

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/MediationSuccessfulRule.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+@Component
+public class MediationSuccessfulRule {
+    public static final String STAFF_UPLOAD_MEDIATION_AGREEMENT =
+        "Upload Mediation Agreement";
+    public static final String STAFF_UPLOAD_TYPE_MEDIATION_AGREEMENT =
+            "You need to upload a document of type Mediation agreement";
+    public static final String STAFF_UPLOAD_PDF_MEDIATION_AGREEMENT =
+            "Mediation agreement needs to be of type PDF";
+
+    public List<String> validateMediationAgreementUploadedByCaseworker(CCDCase ccdCase) {
```
Comment (andrewwa-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/MediationSuccessfulRule.java]: This method could use revisiting. You can get different results based on the order the documents are in.

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/MediationSuccessfulRuleTest.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    private final MediationSuccessfulRule mediationSuccessfulRule = new MediationSuccessfulRule();
+
+    @Test
+    public void shouldReturnValidationMessageWhenNoDocumentIsUploaded() {
+        CCDCase ccdCase = CCDCase.builder().build();
+
+        List<String> validations = mediationSuccessfulRule.validateMediationAgreementUploadedByCaseworker(ccdCase);
+
+        Assertions.assertThat(validations).isNotEmpty()
+                .hasSize(1)
```
Comment (andrewwa-kainos) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/MediationSuccessfulRuleTest.java]: `.isNotEmpty()` is implied by `.hasSize(1)` - no need to repeat

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/MediationSuccessfulRuleTest.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
+package uk.gov.hmcts.cmc.claimstore.services.ccd.callbacks.rules;
+
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+import uk.gov.hmcts.cmc.ccd.domain.CCDCase;
+import uk.gov.hmcts.cmc.ccd.sample.data.SampleData;
+
+import java.util.List;
+
+public class MediationSuccessfulRuleTest {
```
Comment (andrewwa-kainos) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/MediationSuccessfulRuleTest.java]: Please add tests with multiple documents to verify their order does not make a difference

Comment (Valkrys) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/MediationSuccessfulRuleTest.java]: will do

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/MediationSuccessfulRuleTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/MediationSuccessfulRule.java  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4957521&bid=16382220)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/MediationSuccessfulRuleTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/MediationSuccessfulRule.java  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4957521&bid=16382220)

**Comment by timja** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/test/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/MediationSuccessfulRuleTest.java  1
- src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/callbacks/rules/MediationSuccessfulRule.java  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/HMCTS/cmc-claim-store/pullRequest?prid=4957521&bid=16382220)

**Pull request closed without merging**